### PR TITLE
Do not fail ready nodes w/o recent config requests

### PR DESCRIPTION
**Pull request by hakonhall**
This code was used to support non-Docker tenant hosts, but now only affects
ready cfg and proxy containers which may not even exist and cannot possibly
issue config requests (when in ready).

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java [hunk 1]
```diff
@@ -336,6 +332,9 @@ private boolean throttle(Node node) {
                                                .matching(n -> n.history().hasEventAfter(History.Event.Type.failed,
                                                                                         startOfThrottleWindow));
 
+        log.info("node = " + node + ", recentlyFailedNodes.size() = " + recentlyFailedNodes.size() +
+                 ", throttlePolicy.allowedToFailOf(" + allNodes.size() + ") = " +
+                 throttlePolicy.allowedToFailOf(allNodes.size()));
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java:337]:
```suggestion
```

**Review by freva — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java:337]: Thank you

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailerTest.java [hunk 2]
```diff
@@ -627,47 +589,11 @@ public void failing_divergent_ready_nodes() {
...
-            }
-            tester.runMaintainers();
-            assertEquals(4, tester.nodeRepository.nodes().list(Node.State.failed).size());
-
-            // 24 more hours pass, nothing happens
-            for (int minutes = 0, interval = 30; minutes < 24 * 60; minutes += interval) {
-                tester.clock.advance(Duration.ofMinutes(interval));
-                tester.allNodesMakeAConfigRequestExcept(deadNodes);
-            }
+            // TODO: Remove deadNodes
```
Comment (hakonhall) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailerTest.java:595]: I'll do this now...

Comment (hakonhall) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailerTest.java]: done

Comment (freva) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailerTest.java]: :100:

**Pull request merged**